Migrated circle CI config to 2.0

sbuliarca opened the pull request:
This version is close to the one in  [cookiecuter from v1](https://github.com/Financial-Times/cookiecutter-upp-golang/blob/master/%7B%7B%20cookiecutter.repo_name%20%7D%7D/circle.yml) but it doesn't use govendor for fetching dependencies, as vendoring is not introduced for this repo yet.

Commit:
Migrated circle ci config to 2.0 that is close to the cookiecutter version

Fixed tests

Commit:
Added names to commands

nwrigh commented on .circleci/config.yml:
```diff
@@ -0,0 +1,59 @@
+jobs:
+  build:
+    docker:
+      - image: circleci/golang:1.7
```
This should be 1.8

sbuliarca replied:
Thanks, fixed.

nwrigh commented on .circleci/config.yml:
```diff
@@ -0,0 +1,59 @@
+
+      - store_artifacts:
+          path: /tmp/artifacts
+          destination: build
```
We can also to a build of the docker image:

  docker_build:
    working_directory: /public-concordances-api
    docker:
      - image: docker:1.12.6-git
    steps:
      - checkout
      - setup_docker_engine
      - run:
          name: Build Dockerfile
          command: docker build .

sbuliarca replied:
Cool, I added it here also.

nwrigh commented on .circleci/config.yml:
```diff
@@ -0,0 +1,59 @@
+      - run:
+          name: "Build project"
+          command: |
+            go build -v
```
> govendor build -v 

instead?

sbuliarca replied:
Sure.

nwrigh commented on .circleci/config.yml:
```diff
@@ -0,0 +1,59 @@
+          # Not using govendor for syncing deps as vendoring is not yet introduced for this repository
+          command: |
+            go get -u github.com/kardianos/govendor
+            # govendor sync
```
Why not govendor sync?

sbuliarca replied:
I introduced vendoring also, so now we can use govendor.

Commit:
Introduced govendoring
Adjusted files to match the cookiecutter.
Added building the docker image in circle ci

Commit:
Removed go get from Dockerfile... all deps are managed with govendoring

Commit:
Trying to fix building on circleci

Commit:
Fixed vendoring locations.

Commit:
Added missing locations

sbuliarca commented:
Closing this one, as I realized vendoring was already introduced on master.
